libs/threading_base/include/hpx/threading_base/register_thread.hpp

```diff
@@ -93,6 +93,22 @@ namespace hpx { namespace threads {
         };
     }    // namespace detail
 
+    template <typename F>
+    auto make_thread_function(F&& f)
```

Comment:
Could you write this as `decltype(auto)` please?

Reply:
Mostly for consistency/cosmetics, `auto` yields the same result in this case.

Reply:
It might be evenbetter to write this as:
```
template <typename F>
thread_function_type make_thread_function(F&& f)
{
    return {detail::thread_function<typename std::decay<F>::type>{
        std::forward<F>(f)}};
}
```

Reply:
The second option looks good to me. I'll change that.